ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/MixedTableMaintainer.java

```diff
@@ -0,0 +1,286 @@
+      changeFiles = IcebergTableUtil.getAllContentFilePath(arcticTable.asKeyedTable().changeTable());
+      baseFiles = IcebergTableUtil.getAllContentFilePath(arcticTable.asKeyedTable().baseTable());
+    } else {
+      baseMaintainer = new BaseTableMaintainer(arcticTable.asUnkeyedTable());
+      changeFiles = new HashSet<>();
+      baseFiles = IcebergTableUtil.getAllContentFilePath(arcticTable.asUnkeyedTable());
+    }
+
+    if (TableTypeUtil.isHive(arcticTable)) {
+      hiveFiles = HiveLocationUtil.getHiveLocation(arcticTable);
+    } else {
+      hiveFiles = new HashSet<>();
+    }
+  }
+
+  @Override
+  public void cleanContentFiles(long lastTime) {
+    if (changeMaintainer != null) {
+      changeMaintainer.cleanContentFiles(lastTime);
+    }
+    baseMaintainer.cleanContentFiles(lastTime);
+  }
+
+  @Override
+  public void cleanMetadata(long lastTime) {
+    if (changeMaintainer != null) {
+      changeMaintainer.cleanMetadata(lastTime);
+    }
+    baseMaintainer.cleanMetadata(lastTime);
+  }
+
+  @Override
+  public void cleanDanglingDeleteFiles() {
+    if (changeMaintainer != null) {
+      changeMaintainer.cleanDanglingDeleteFiles();
+    }
+    baseMaintainer.cleanDanglingDeleteFiles();
+  }
+
+  @Override
+  public void expireSnapshots(TableRuntime tableRuntime) {
+    if (changeMaintainer != null) {
+      changeMaintainer.expireSnapshots(tableRuntime);
+    }
+    baseMaintainer.expireSnapshots(tableRuntime);
+  }
+
+  @Override
+  public void expireSnapshots(long mustOlderThan) {
+    if (changeMaintainer != null) {
+      changeMaintainer.expireSnapshots(mustOlderThan);
+    }
+    baseMaintainer.expireSnapshots(mustOlderThan);
+  }
+
+  public ChangeTableMaintainer getChangeMaintainer() {
+    return changeMaintainer;
+  }
+
+  public BaseTableMaintainer getBaseMaintainer() {
+    return baseMaintainer;
+  }
+
+  @SafeVarargs
+  private final Set<String> mergeSets(Set<String>... sets) {
+    Set<String> result = new HashSet<>();
+    for (Set<String> set : sets) {
+      result.addAll(set);
+    }
+    return result;
+  }
+
+  public class ChangeTableMaintainer extends IcebergTableMaintainer {
+
+    private static final int DATA_FILE_LIST_SPLIT = 3000;
+
+    private final UnkeyedTable unkeyedTable;
+
+    public ChangeTableMaintainer(UnkeyedTable unkeyedTable) {
+      super(unkeyedTable);
+      this.unkeyedTable = unkeyedTable;
+    }
+
+    @Override
+    public Set<String> orphanFileCleanNeedToExcludeFiles() {
+      return mergeSets(changeFiles, baseFiles, hiveFiles);
+    }
+
+    public void expireSnapshots(TableRuntime tableRuntime) {
+      expireSnapshots(Long.MAX_VALUE);
+    }
+
+    @Override
+    public void expireSnapshots(long mustOlderThan) {
+      long changeDataTTL = getChangeDataTTL();
+      expireFiles(Longs.min(System.currentTimeMillis() - changeDataTTL, mustOlderThan));
+      super.expireSnapshots(mustOlderThan);
```

Comment:
Could use changeDataTTL directly here so that it can reduce a call for `getChangeDataTTL()`  in `olderThanSnapshotNeedToExpire(long mustOlderThan)`?